experiment/simulation/js/add_numbers.js

```diff
@@ -0,0 +1,339 @@
+            document.getElementById(`basecase-return-${language1}`).value = basecaseReturnSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let recursecaseSelect = document.getElementById(`recursecase-${language}`);
+    recursecaseSelect.addEventListener('change', function () {
+        recursecase = recursecaseSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`recursecase-${language1}`).value = recursecaseSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let variableSelect = document.getElementById(`variable-${language}`);
+    variableSelect.addEventListener('change', function () {
+        variable = variableSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`variable-${language1}`).value = variableSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let operatorSelect = document.getElementById(`operator-${language}`);
+    operatorSelect.addEventListener('change', function () {
+        operator = operatorSelect.value;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`operator-${language1}`).value = operatorSelect.value;
+        })
+    })
+});
+languages.forEach(language => {
+    let nSelect = document.getElementById(`n-${language}`);
+    nSelect.addEventListener('change', function () {
+        maxStackDepth = parseInt(nSelect.value);
+        stackText.textContent = maxStackDepth;
+        reset();
+
+        languages.forEach(language1 => {
+            document.getElementById(`n-${language1}`).value = nSelect.value;
+        })
+    })
+});
+
+function step() {
+    executedsteps++;
+    if (stepstoexecute != -1 && executedsteps > stepstoexecute)
+        return false;
+    return true;
+}
+
+function changeOperator(operator) {
+    switch (operator) {
+        case 'eq':
+            return '==';
+        case 'neq':
+            return '!=';
+        case 'lt':
+            return '<';
+        case 'le':
+            return '<=';
+        case 'gt':
+            return '>';
+        case 'ge':
+            return '>=';
+        default:
+            return '';
+    }
+}
+
+function reset() {
+    stackElement.innerHTML = '';
+    stackElement.style.color = 'black';
+    stackHTML = '';
+
+    resultContent = '';
+    resultElement.style.color = 'black';
+    resultElement.textContent = ' ';
+
+    runButton.disabled = false;
+    nextButton.disabled = false;
+    prevButton.disabled = true;
+
+    stepstoexecute = -1;
+    executedsteps = 0;
+}
+
+function sumOfNNaturalNumbers(n) {
+    if (n <= 0) {
+        return "Please enter a positive integer.";
+    }
+
+    const sum = (n * (n + 1)) / 2;
+    return sum;
+}
+
+function step_sumOfNNaturalNumbers(n, stackDepth) {
+    newHTML = `<div class="stack">
+        stackDepth: ${stackDepth}<br>
+        n: ${eval(n)}<br>
```

Comment:
using eval is considered a bad practice since it introduces security issues. Please look [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) and try to find alternatives to eval

Reply:
As said in the article, have added strict mode for eval, and removed from some places, where it was not necessary, but issue is that without it, would need to add several layers of if-else-if blocks to tackle combinations of all the options given to user, which in itself would make the code very long.

Reply:
Also, the variables used in eval are either declared in the js file, or preprocessed from the options from the html, and no direct input is taken from the user, so shouldn't be an issue.

Couldn't find anything equivalent of eval to use.

Reply:
ok, I don't want to get into too many details so try to keep its use at strictly essential places

Reply:
Yes sir, at max 2 places in a js file, where couldn't solve the problem using any other thing.